sdk/python/feast/repo_operations.py

```diff
@@ -172,6 +172,15 @@ def parse_repo(repo_root: Path) -> RepoContents:
                 assert stream_source
                 if not any((stream_source is ds) for ds in res.data_sources):
                     res.data_sources.append(stream_source)
+            elif isinstance(obj, BatchFeatureView) and not any(
```

Comment:
add a test for correctly applying this?

would probably do it in `unit/local_feast_tests/` and just create an `example_feature_repo` and load it with `get_example_repo`?